pkg/broker/config/volume/volume_test.go

```diff
@@ -154,12 +158,31 @@ func TestSyncConfigFromFile(t *testing.T) {
 
 	delete(data.Brokers["ns2/broker2"].Targets, "name4")
 	b, _ = proto.Marshal(data)
-	ioutil.WriteFile(tmp.Name(), b, 0644)
+	atomicWriteFile(t, dir, tmp.Name(), b)
 
 	<-ch
 
 	gotTargets = targets.(*Targets).Load()
 	if !proto.Equal(data, gotTargets) {
-		t.Errorf("updated targets got=%+v, want%+v", gotTargets, data)
+		t.Errorf("updated targets got=%+v, want=%+v", gotTargets, data)
+	}
+}
+
+func atomicWriteFile(t *testing.T, tempDir, file string, bytes []byte) {
+	t.Helper()
+	// In order to more closely replicate how K8s writes ConfigMaps to the file system, we will
+	// atomically swap out the file by writing it to a temp directory, then renaming it into the
+	// directory we are watching.
+	tmpFile, err := ioutil.TempFile(tempDir, "temp-*")
```

Comment:
My preference would be to write a different directory, as that will more closely match what k8s does. In particular, depending on how the file system watcher is setup, this approach will cause multiple change notifications (ie temp file created, temp file written, overwrite config file).